src/test/java/io/confluent/connect/jdbc/JdbcSourceTaskUpdateTest.java

```diff
@@ -178,6 +179,40 @@ public void testTimestamp() throws Exception {
   }
 
   @Test
+  public void testTimestampWithDelay() throws Exception {
+    expectInitializeNoOffsets(Arrays.asList(SINGLE_TABLE_PARTITION));
+
+    PowerMock.replayAll();
+
+    // Manage these manually so we can verify the emitted values
+    db.createTable(SINGLE_TABLE_NAME,
+            "modified", "TIMESTAMP NOT NULL",
+            "id", "INT");
+
+    db.insert(SINGLE_TABLE_NAME, "modified", JdbcUtils.formatUTC(new Timestamp(10L)), "id", 1);
+
+    startTask("modified", null, null, 4L);
+    verifyTimestampFirstPoll(TOPIC_PREFIX + SINGLE_TABLE_NAME);
+
+    Long currentTime = new Date().getTime();
```

Comment:
The delay will be relative to the time on the DB (which, in the test, will be local time and not 1970), so we still need timestamps in the records that are more recent.